src/script/interpreter.cpp

```diff
@@ -1674,8 +1676,7 @@ bool VerifyScript(const Span<const unsigned char>& scriptSig, const Span<const u
         if (flags & SCRIPT_VERIFY_WITNESS) {
             if (IsWitnessProgram(redeem_script, witnessversion, witnessprogram)) {
                 hadWitness = true;
-                const CScript expected_scriptsig = CScript() << redeem_script;
-                if (scriptSig != expected_scriptsig) {
+                if (scriptSig.size() != redeem_script.size() + 1 || scriptSig[0] != redeem_script.size()) {
```

Comment:
Strictly speaking that's not semantically equivalent to the code replaced. Before it would compare `scriptSig` and the pushed `redeem_script` byte-by-byte, now with the change it only compares that the size matches and the first byte (the push).
I guess that is indended though, as `redeem_script` is derived from `scriptSig` above anyways, and hence the content comparison is not needed?